inc/ticketconfig.class.php

```diff
@@ -0,0 +1,318 @@
+ * -------------------------------------------------------------------------
+ */
+
+if (!defined('GLPI_ROOT')) {
+   die("Sorry. You can't access directly to this file");
+}
+
+class PluginCreditTicketConfig extends CommonDBTM {
+
+   public static $rightname = 'entity';
+
+   static function getTypeName($nb = 0) {
+      return _n('Default credit options', 'Credit vouchers', $nb, 'credit');
+   }
+
+   static function createTicketOption(Ticket $ticket) {
+      $ticketConfig = new PluginCreditTicketConfig();
+      $data = [
+         "tickets_id"               => $ticket->getID(),
+         "credit_default_followup"  => 0,
+         "credit_default_task"      => 0,
+         "credit_default_solution"  => 0,
+      ];
+      $ticketConfig->add($data);
+      return $ticketConfig;
+   }
+
+   public function calculateGlobalDefault() {
+      if (!isset($_SESSION['credit']['post_update'])) {
+         if (($this->fields['credit_default_followup'] ===  $this->fields['credit_default_task'])
+         && ($this->fields['credit_default_task'] ===  $this->fields['credit_default_solution'])) {
+            $this->fields['credit_default'] = $this->fields['credit_default_solution'];
+         } else {
+            $this->fields['credit_default'] = 0;
+         }
+         $_SESSION['credit']['post_update'] = true;
+         $this->update($this->fields);
+      } else {
+         unset($_SESSION['credit']['post_update']);
+      }
+   }
+
+   public function post_updateItem($history = 1) {
+      $this->calculateGlobalDefault();
+   }
+
+   public function post_addItem($history = 1) {
+      $this->calculateGlobalDefault();
+   }
+
+   /**
+    * Get default credit for entity and itemtype
+    *
+    * @param $ID           integer     entities ID
+    * @param $start        integer     first line to retrieve (default 0)
+    * @param $limit        integer     max number of line to retrieve (0 for all) (default 0)
+    * @param $sqlfilter    string      to add a SQL filter (default '')
+    * @return array of vouchers
+   **/
+   static function getDefaultForTicket($ID, $itemtype) {
+      $ticketConfig = new PluginCreditTicketConfig();
+      if ($ticketConfig->getFromDBByCrit(["tickets_id" => $ID])) {
+         switch ($itemtype) {
+            case ITILFollowup::getType():
+               return $ticketConfig->fields['credit_default_followup'];
+               break;
+
+            case TicketTask::getType():
+               return $ticketConfig->fields['credit_default_task'];
+               break;
+
+            case ITILSolution::getType():
+               return $ticketConfig->fields['credit_default_solution'];
+               break;
+         }
+      }
+      return 0;
+   }
+
+
+   /**
+    * Show default credit option ticket
+    *
+    * @param $ticket Ticket object
+   **/
+   static function showForTicketTab(Ticket $ticket, $isTicket = false) {
+
+      if (!Session::haveRight("entity", UPDATE)) {
```

Comment:
Could be better to be sure that we won't forget to update it if we define a dedicated rightname.
```suggestion
      if (!Session::haveRight(PluginCreditEntityConfig::$rightname, UPDATE)) {
```